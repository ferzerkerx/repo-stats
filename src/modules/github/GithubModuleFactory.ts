import { CollectorModuleFactory, Utils } from '../../metrics';
import { GithubCollectorConfig, GithubMetricItem } from './collector/Types';
import { GithubService } from './Types';
import { GithubClientImpl } from './GithubClientImpl';
import { GithubServiceImpl } from './GithubServiceImpl';
import { GithubCollectorService } from './collector/GithubCollectorService';
import { GithubRepositoryImpl } from './GithubRepositoryImpl';

export class GithubModuleFactory
  implements CollectorModuleFactory<GithubCollectorConfig, GithubMetricItem> {
  private static githubService(): GithubService {
    Utils.checkEnvVar('GITHUB_TOKEN');
    const githubClient = new GithubClientImpl({
      token: `${process.env.GITHUB_TOKEN}`
    });
    const githubRepository = new GithubRepositoryImpl(githubClient);
    return new GithubServiceImpl(githubRepository);
  }

  collectorInstance(): GithubCollectorService {
    return new GithubCollectorService(GithubModuleFactory.githubService());
  }

  collectorConfiguration(obj: any): GithubCollectorConfig {
    return new GithubCollectorConfig(obj);
  }
}
