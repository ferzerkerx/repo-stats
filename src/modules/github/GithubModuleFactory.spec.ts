import { GithubModuleFactory } from './GithubModuleFactory';

describe('GithubModuleFactory', () => {
  const githubModuleFactory: GithubModuleFactory = new GithubModuleFactory();

  beforeEach(() => {
    jest.resetModules();
    delete process.env.GITHUB_TOKEN;
  });

  it('should create collector', () => {
    process.env.GITHUB_TOKEN = 'SOME_TOKEN';
    const collectorInstance = githubModuleFactory.collectorInstance();
    expect(collectorInstance).not.toBeNull();
  });

  it('should fail to create collector', () => {
    expect(
      githubModuleFactory.collectorInstance
    ).toThrowErrorMatchingSnapshot();
  });

  it('should create collectorConfiguration', () => {
    const collectorConfiguration = githubModuleFactory.collectorConfiguration(
      {}
    );
    expect(collectorConfiguration).not.toBeNull();
  });
});
