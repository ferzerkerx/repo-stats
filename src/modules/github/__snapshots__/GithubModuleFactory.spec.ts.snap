// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubModuleFactory should fail to create collector 1`] = `"env.GITHUB_TOKEN not set!"`;
